### PR TITLE
Processing data

### PR DESCRIPTION
**Pull request by isaka-lumato**
### Processing data

- Created models
- Used models to insert data with rails console(`rails c` command)
- Created custom methods

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -1,2 +1,11 @@
 class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+  after_save :update_comment_counter
+
+  private
+
+  def update_comment_counter
+    post.increment!(:comments_counter)
+  end
```
Comment (rotimiazeez) [inline app/models/comment.rb:10]: - [ ] Kindly make sure the associations between models are set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).👍
```
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

File: app/models/like.rb [hunk 2]
```diff
@@ -1,2 +1,11 @@
 class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+  after_save :update_likes_count
+
+  private
+
+  def update_likes_count
+    post.increment!(:likes_counter)
+  end
```
Comment (rotimiazeez) [inline app/models/like.rb:10]: - [ ] Kindly make sure the associations between models are set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).👍
```
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

File: app/models/post.rb [hunk 3]
```diff
@@ -1,2 +1,16 @@
...
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+  after_save :update_post_counter
+
+  def recent_comments
+    comments.limit(5).order(created_at: :desc)
+  end
+
+  private
```
Comment (rotimiazeez) [inline app/models/post.rb:11]: - [ ] Kindly make sure the associations between models are set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).👍
```
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
Hi @isaka-lumato , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BrianSammit — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello @isaka-lumato, 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

TO HIGHLIGHT:

- Your project is professional 
- Your repo is professional 
- Well structured files 


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

Well-Done-Applause


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**